internal/envoy/bootstrap.go

```diff
@@ -102,6 +104,18 @@ func (c *BootstrapConfig) GetAdminAccessLogPath() string {
 func (c *BootstrapConfig) GetDNSLookupFamily() string {
 	return stringOrDefault(c.DNSLookupFamily, "auto")
 }
+
+// ValidAdminAddress checks if the address supplied is
+// "localhost" or an IP address. Only a Unix Socket
+// is supported for this address to mitigate security.
+func ValidAdminAddress(address string) error {
+	// Value of "localhost" is invalid.
+	if address == "localhost" || net.ParseIP(address) != nil {
+		return fmt.Errorf("invalid value %q, cannot be `localhost` or an ip address", address)
+	}
+	return nil
```

Comment:
Should we check that the unix socket exists, or at least that the path is a valid one?

Reply:
I'm not sure of the timing of that type of check. This is the bootstrap code which executes in the initContainer. Then the pod would startup after that. Need to double check if the volumes are all ready in init phase or not, I don't remember off the top of my head.